plugins/nanospell/plugin.js

```diff
@@ -159,6 +164,12 @@
 		init: function (editor) {
 			var self = this;
 
+			// a lock to prevent multiple spellchecks
+			this._isBusy = false;
+
+			// store the current timer
+			this._timer = null;
```

Comment:
Hmm I've looked into them.  Definitely needs a few more refactors to get there though.  That would be cool.